src/bson.rs

```diff
@@ -304,6 +304,18 @@ impl From<DbPointer> for Bson {
     }
 }
 
+impl<T> From<Option<T>> for Bson
+where
+    T: Into<Bson>,
+{
+    fn from(a: Option<T>) -> Bson {
+        match a {
+            None => Bson::Null,
+            Some(t) => T::into(t),
```

Comment:
It's a little more idiomatic to just invoke `into` as a method on `t` here: `t.into()`